config.go

```diff
@@ -199,5 +217,13 @@ func validateConfig(c *Config) error {
 		return ErrConfigInvalidLogger
 	}
 
+	if c.iteratorType != "" && c.iteratorType != "TRIM_HORIZON" && c.iteratorType != "LATEST" && c.iteratorType != "AT_TIMESTAMP" {
```

Comment:
With an enum type you can possibly do something like:

```
type IteratorType int

const (
    TrimHorizon IteratorType = iota
    Latest
    AtTimestamp
)

func (it IteratorType) String() (string, error) {
    switch it {
    case TrimHorizon:
        return "TRIM_HORIZON", nil
    case Latest:
        return "LATEST", nil
    case AtTimestamp:
        return "AT_TIMESTAMP", nil
    default:
        return "", ErrConfigInvalidIteratorType
    }
}
```

Reply:
Equally you could also leverage the types already defined? https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/kinesis@v1.40.1/types#ShardIteratorType